CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+# How to contribute
+
+## Voter dApp
+
+This is a browser client that UMA token holders can use to vote on price requests. The dApp interacts with the deployed [DVM](https://docs.umaproject.org/uma/oracle/technical_architecture.html) code using MetaMask's browser extension. The source code was bootstrapped from [Create React App](https://github.com/facebook/create-react-app).
+
+Features currently include:
+
+- Deploy a [designated voting proxy contract](https://docs.umaproject.org/uma/oracle/voting_with_UMA_2-key_contract.html)
+- Commit an encrypted vote
+- Backup salts associated with latest encrypted votes
+- Reveal an encrypted vote
+- Retrieve rewards for resolved votes
+- View results of resolved votes
+
+### Making changes to the Voter dApp
+
+The source code for this React app can be found in `./voter-dapp/src/`. At a minimum, you must ensure that a user can go through a simple commit and reveal flow, both using a voting proxy and not using one.
+
+#### Steps to generate price requests
+
+0. Change your directory to `./protocol/`
+1. Start a local blockchain on port `9545` with enough starting ETH balances and a high enough gas limit:
+   ```sh
+   ganache-cli -p 9545 -e 10000000000 -l 9000000
+   ```
+   This seeds every account with `10000000000` ETH and sets the block gas limit to 9 million wei.
+1. Open another window and deploy all contracts:
+   ```sh
+   cd core && $(npm bin)/truffle migrate --reset --network test
+   ```
+1. Make a price request for the "BTC/USD" identifier at the timestamp `1570000000`. Note that the timestamp passed in denotes the Unix Epoch time. This script will take care of registering the chosen price identifier and requesting a price:
+   ```sh
+   $(npm bin)/truffle exec ./scripts/local/RequestOraclePrice.js --network test --identifier BTC/USD --time 1570000000
+   ```
+1. Advance time to the next voting round's commit phase so that the price request becomes available to vote on. Note that you must run this script twice, once to advance to the current round's reveal phase, and once more to advance to the next round's commit phase. The default starting phase is the commit phase:
+   ```sh
+   $(npm bin)/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
+   $(npm bin)/truffle exec ./scripts/local/AdvanceToNextVotingPhase.js --network test
+   ```
+
+At this point, the price request is ready to be voted on.
+
+### Steps to vote on price requests through the dApp
+
+0. Open a new console tab and change your directory to `./protocol/voter-dapp` via `cd ../voter-dapp`. At this point, you should have three console tabs open: one that is running `ganache-cli` locally, one for running helper scripts from the previous section, and one to run the voter dapp React app.
+1. Start the application: `npm start`
```

Comment:
if they have not done so already you need to run an `npm install`

Reply:
ah I assume they would have built already. I'll link to the prereq's site on the doc site